t/api/09-comments.t

```diff
@@ -0,0 +1,123 @@
+# Copyright (C) 2016 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+BEGIN {
+    unshift @INC, 'lib';
+}
+
+use Mojo::Base -strict;
+use Test::More;
+use Test::Mojo;
+use Test::Warnings;
+use OpenQA::Test::Case;
+use OpenQA::Client;
+use Mojo::IOLoop;
+
+OpenQA::Test::Case->new->init_data;
+
+my $t = Test::Mojo->new('OpenQA::WebAPI');
+
+# XXX: Test::Mojo loses it's app when setting a new ua
+# https://github.com/kraih/mojo/issues/598
+my $app = $t->app;
+$t->ua(OpenQA::Client->new(apikey => 'PERCIVALKEY02', apisecret => 'PERCIVALSECRET02')->ioloop(Mojo::IOLoop->singleton));
+$t->app($app);
+
+sub test_get_comment {
+    my ($in, $id, $comment_id, $supposed_text) = @_;
+    my $get = $t->get_ok("/api/v1/$in/$id/comments/$comment_id");
+    is($get->tx->res->json->{text}, $supposed_text, 'comment text is correct');
+}
+
+sub test_get_comment_invalid_job_or_group {
+    my ($in, $id, $comment_id) = @_;
+    my $get = $t->get_ok("/api/v1/$in/$id/comments/$comment_id")->status_is(404, 'comment not found');
+    like($get->tx->res->json->{error}, qr/$id does not exist/, 'job/group does not exist');
+}
+
+sub test_get_comment_invalid_comment {
+    my ($in, $id, $comment_id) = @_;
+    my $get = $t->get_ok("/api/v1/$in/$id/comments/$comment_id")->status_is(404, 'comment not found');
+    is($get->tx->res->json->{error}, "Comment $comment_id does not exist", 'comment does not exist');
```

Comment:
test also 404 for delete or create - because I think your code is buggy in these cases.